src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -516,6 +534,29 @@ private BoundPattern BindDeconstructionPattern(DeconstructionPatternSyntax node,
                 deconstruction: patterns.ToImmutableAndFree(), propertiesOpt: propertiesOpt, variable: variableSymbol, variableAccess: variableAccess, hasErrors: hasErrors);
         }
 
+        /// <summary>
+        /// Check that the given name designates a tuple element at the given index, and return that element.
+        /// </summary>
+        private FieldSymbol CheckIsTupleElement(SyntaxNode node, NamedTypeSymbol tupleType, string name, int tupleIndex, DiagnosticBag diagnostics)
```

Comment:
Are you sure you need to return the `FieldSymbol`?
If not, you could simply check that `tupleType.TupleElementNames[tupleIndex]` matches `name`.

Reply:
Cannot check the name, as you are allowed to use any name for that position's element (e.g. `Item2` instead of `LastName`)